spot_wrapper/wrapper.py

```diff
@@ -729,252 +474,228 @@ def __init__(
+                        e, sleep_secs
                     )
                 )
+                time.sleep(sleep_secs)
 
-            # Build image requests by camera
-            self._image_requests_by_camera = {}
-            for camera in IMAGE_SOURCES_BY_CAMERA:
-                if camera == "hand" and not self._robot.has_arm():
-                    continue
-                self._image_requests_by_camera[camera] = {}
-                image_types = IMAGE_SOURCES_BY_CAMERA[camera]
-                for image_type in image_types:
-                    if image_type.startswith("depth"):
-                        image_format = image_pb2.Image.FORMAT_RAW
-                        pixel_format = image_pb2.Image.PIXEL_FORMAT_DEPTH_U16
-                    else:
-                        image_format = image_pb2.Image.FORMAT_JPEG
-                        if camera == "hand" or self._rgb_cameras:
-                            pixel_format = image_pb2.Image.PIXEL_FORMAT_RGB_U8
-                        elif camera != "hand":
-                            self._logger.info(
-                                f"Switching {camera}:{image_type} to greyscale image format."
-                            )
-                            pixel_format = image_pb2.Image.PIXEL_FORMAT_GREYSCALE_U8
-
-                    source = IMAGE_SOURCES_BY_CAMERA[camera][image_type]
-                    self._image_requests_by_camera[camera][
-                        image_type
-                    ] = build_image_request(
-                        source,
-                        image_format=image_format,
-                        pixel_format=pixel_format,
-                        quality_percent=75,
-                    )
+        # Core Async Tasks
+        self._async_task_list = []
+        self._robot_state_task = AsyncRobotState(
+            self._robot_state_client,
+            self._logger,
+            max(0.0, self._rates.get("robot_state", 0.0)),
+            self._callbacks.get("robot_state", None),
+        )
+        self._robot_metrics_task = AsyncMetrics(
+            self._robot_state_client,
+            self._logger,
+            max(0.0, self._rates.get("metrics", 0.0)),
+            self._callbacks.get("metrics", None),
+        )
+        self._lease_task = AsyncLease(
+            self._lease_client,
+            self._logger,
+            max(0.0, self._rates.get("lease", 0.0)),
+            self._callbacks.get("lease", None),
+        )
+        self._idle_task = AsyncIdle(
+            self._robot_command_client, self._logger, 10.0, self
+        )
+        self._estop_monitor = AsyncEStopMonitor(
+            self._estop_client, self._logger, 20.0, self
+        )
 
-            # Store the most recent knowledge of the state of the robot based on rpc calls.
-            self._init_current_graph_nav_state()
+        self._estop_endpoint = None
+        self._estop_keepalive = None
+        self._robot_id = None
```

Comment:
The property down below uses the robot_params dict instead of self._robot_id? If this isn't used anywhere they it should be removed to avoid confusion

Reply:
I've made it so that the wrapper class no longer uses the convenience dict for accessing these values or returning them from properties. It still has the dict to pass to things which it instantiates.